pool/src/pool.rs

```diff
@@ -152,6 +152,35 @@ impl Pool {
 		Ok(Some(tx))
 	}
 
+	/// evict last 10% of the transaction from the pool (based on fees over weight)
+	pub fn evict_from_pool(&mut self, max_pool_size: usize) {
+		let tx_buckets = self.bucket_transactions();
```

Comment:
Not directly related to this PR but I wonder how well `bucket_transactions()` handles the case where a tx in the pool is dependent on multiple prior unconfirmed txs.

Reply:
Just thought of something else - not sure we want to evict the whole "bucket" of txs either - it may make sense to just evict the dependent txs?

Reply:
Buckets are all dependent transactions. Independent transactions are all on one-buckets and if 5 transactions depend on each other, no matter which way, they'll all end up in the same bucket.

From a transaction selection standpoint, the nice thing is we already have child-pays-for-parent. The bad thing is someone involved in a transaction could try to kick it out of the pools by appending a very large one. Little risky and costly attack but still a current weakness.